src/main/java/com/matera/cadastrocentral/client/ClientService.java

```diff
@@ -0,0 +1,89 @@
+package com.matera.cadastrocentral.client;
+
+import com.matera.cadastrocentral.identitydocument.IdentityDocumentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ClientService {
+
+    private final ClientRepository clientRepository;
+    private final IdentityDocumentRepository identityDocumentRepository;
+
+    @Autowired
+    public ClientService(final ClientRepository clientRepository,
+                         final IdentityDocumentRepository identityDocumentRepository) {
+        this.clientRepository = clientRepository;
+        this.identityDocumentRepository = identityDocumentRepository;
+    }
+
+    /* API requests */
+
+    // 1. Get all clients from the database.
+    public List<Client> getAllClients() {
+        return clientRepository.findAll();
+    }
+
+    // 2. Get a specific client by id.
+    public Optional<Client> getClientById(UUID clientId) {
+        Optional<Client> client = clientRepository.findById(clientId);
+        if(client.isPresent()){
+            return client;
+        } else {
+            throw new ResponseStatusException(
+                    HttpStatus.BAD_REQUEST,
+                    "Client ID does not exist in the database! Try a valid one."
+            );
+        }
+    }
+
+    // 3. Insert a client into the database.
+    public Client insertClient(final ClientDTO clientDTO) {
+        Client client = clientRepository.save(new Client(clientDTO));
+        client.getIdentityDocumentEntityList().forEach(
+                identityDocumentEntity -> {
+                    identityDocumentEntity.setClient(client);
+                    identityDocumentRepository.save(identityDocumentEntity);
+                }
+        );
+        return client;
+    }
+
+    // 4. Alter a client information in the database.
+    public Client alterClient(UUID clientId, ClientDTO clientDTO) {
+        try {
+            Client alteredClient = clientRepository.getOne(clientId);
+            alteredClient.setName(Optional.ofNullable(
```

Comment:
you can skip using Optional as validation here since these fields are @NotNull on DTO class

Reply:
I did some tests here, and in fact he accepts a request with an empty body, or just with the marital_status entity without the name, or just with the name without the marital_status entity. I believe that this is an acceptable behavior, since the client may want to change only some of his data.

But a doubt that came to me was that, the client's DTO the attributes are @NotNull, as you said, however, he can receives DTOs with empty attributes, as I said. Is this right? So, this constraint of attributes only works when I instantiate a DTO?

Reply:
I was reviewing Francisco's PR, and found out about PATCH rs My PUT method is actually working as a PATCH, right? Would it be better to make a PUT that only accepts the full DTO, and a separate PATCH that accepts part of the payload?

Reply:
We made a separate card to this task: https://trello.com/c/ZpoQTEww